code/modules/antagonists/wizard/equipment/soulstone.dm

```diff
@@ -9,11 +9,15 @@
 	desc = "A fragment of the legendary treasure known simply as the 'Soul Stone'. The shard still flickers with a fraction of the full artefact's power."
 	w_class = WEIGHT_CLASS_TINY
 	slot_flags = ITEM_SLOT_BELT
-	var/usability = FALSE
-
 	var/old_shard = FALSE
 	var/spent = FALSE
-	var/purified = FALSE
+	///This controls the color of the soulstone as well as restrictions for who can use it. THEME_CULT is red and is the default of cultist THEME_WIZARD is purple and is the default of wizard and THEME_HOLY is for purified soul stone
+	var/theme = THEME_CULT
+	/// Role check, if any needed
+	var/required_role = /datum/antagonist/cult
+
+/obj/item/soulstone/proc/role_check(mob/who)
+	return required_role ? (who.mind && who.mind.has_antag_datum(required_role, TRUE)) : TRUE
```

Comment:
Can you test this on wizards to ensure that the wizard can still use their spell "Six Soul Stone Shards and the spell Artificer"

Reply:
Fixed in 202c989

For the regular soulstone, the required role is set to check for cultist antag datum. 

For the soulstone/mystic subtype, the required role is redefined to check for wizard antag datum.